go-controller/pkg/node/gateway_init.go

```diff
@@ -189,7 +190,32 @@ func getGatewayNextHops() ([]net.IP, string, error) {
 	if needIPv4NextHop || needIPv6NextHop || gatewayIntf == "" {
 		defaultGatewayIntf, defaultGatewayNextHops, err := getDefaultGatewayInterfaceDetails(gatewayIntf, config.IPv4Mode, config.IPv6Mode)
 		if err != nil {
-			return nil, "", err
+			if !(errors.As(err, new(*GatewayInterfaceMismatchError)) && config.Gateway.Mode == config.GatewayModeLocal && config.Gateway.AllowNoUplink) {
+				return nil, "", err
+			}
+		}
+		if gatewayIntf == "" {
+			if defaultGatewayIntf == "" {
+				return nil, "", fmt.Errorf("unable to find default gateway and none provided via config")
+			}
+			gatewayIntf = defaultGatewayIntf
+		} else {
+			if gatewayIntf != defaultGatewayIntf || len(defaultGatewayNextHops) == 0 {
+				if config.Gateway.Mode == config.GatewayModeLocal && config.Gateway.AllowNoUplink {
+					// For local gw, if not default gateway is available or the provide gateway interface is not the host gateway interface
+					// use nexthop masquerade IP as GR default gw to steer traffic to the gateway bridge
+					if needIPv4NextHop {
+						nexthop := net.ParseIP(types.V4DummyNextHopMasqueradeIP)
+						gatewayNextHops = append(gatewayNextHops, nexthop)
+						needIPv4NextHop = false
+					}
+					if needIPv6NextHop {
+						nexthop := net.ParseIP(types.V6DummyNextHopMasqueradeIP)
+						gatewayNextHops = append(gatewayNextHops, nexthop)
+						needIPv6NextHop = false
+					}
```

Comment:
I don't see it causing any harm. And it can help us debug.

Reply:
If we annotate it, it then becomes part of the API posted for other ovn-k components to use. I guess those components would error out if they did not find the L3 gateway annotation as it was not supported up until now. But now they will have to deal with this special value and interpret it correctly.

So I guess that what you are saying here is that the annotation is not used for anything else other than setting up the default route in the GR.

But it still something future potential consumers of the annotation will need to consider.

Up to you, I just wanted to give this perspective.

Reply:
I didn't think the L3 gateway annotation is part of the API which can be used externally. I thought it is only for OVN-K internal communication. @trozet WDYT?

Reply:
That's what I meant with `part of the API posted for other ovn-k components`, I didn't mean externally although nobody is stopping anyone from consuming that externally as well.

Reply:
Oh, I thought `future potential consumers` means some external components. I think we'd better keep it as it is. Since that is the nexthop we set in the gateway router, it fits the semantics of the annotation name. It a consumer need to deal with this annotation, it shall be aware of what this special value means.

Reply:
we need the L3 nexthops to create the route in NBDB. I'm OK with leaving it annotated. I dont think anything else uses it.

Reply:
Oh, I thought that was happening already for some other reason. I stand corrected!